cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureLoadBalancerMetadataCollector.java

```diff
@@ -39,7 +39,10 @@ private Map<String, Object> parseTargetGroupCloudParams(AuthenticatedContext ac,
         AzureClient azureClient = ac.getParameter(AzureClient.class);
         Map<String, LoadBalancingRule> rules = azureClient.getLoadBalancerRules(resourceGroup, loadBalancerName);
 
+        // rule name ending with special "gateway" suffix is happening _only_ in the case of requested GATEWAY_PRIVATE load balancer
+        // and in that case the second rule is basically the copy of the first one, only the name differs. See arm-v2.ftl
         Map<Integer, String> portToAsMapping = rules.values().stream()
+                .filter(r -> !r.name().endsWith("gateway"))
```

Comment:
maybe it could be a function inside the enum ?

Reply:
the problem is that the enum belongs to the Azure SDK